<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Ryan Sun" />

<meta name="date" content="2024-01-29" />

<title>Generalized Berk-Jones (GBJ) Tutorial</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Generalized Berk-Jones (GBJ) Tutorial</h1>
<h4 class="author">Ryan Sun</h4>
<h4 class="date">2024-01-29</h4>



<p>The GBJ package implements the Generalized Berk-Jones (GBJ) test for
set-based inference in genetic association studies.</p>
<p>Also included in this package are routines to perform the Generalized
Higher Criticism (GHC), Higher Criticism (HC), Berk-Jones (BJ), and
Minimum p-value (minP) tests. Some guidance on choosing between these
methods (which, in principal, test the same null hypothesis and could be
used interchangeably) is also given in the above paper.</p>
<p>The remainder of this vignette provides:</p>
<ul>
<li>A high-level explanation of the problems that GBJ/GHC/HC/BJ/minP can
solve.</li>
<li>A worked example in the context of Genome-Wide Association Studies
(GWAS) where individual-level genotype data is available.</li>
<li>A worked example in the context of GWAS where only single SNP
summary statistics are available.</li>
</ul>
<div id="uses-for-gbj" class="section level2">
<h2>Uses for GBJ</h2>
<p>We find it instructional to begin with a short anecdote from the
class notes of John Tukey (Donoho and Jin, 2004):</p>
<blockquote>
<p>“A young psychologist administers many hypothesis tests as part of a
research project, and finds that, of 250 tests 11 were significant at
the 5% level. The young researcher feels very proud of this fact and is
ready to make a big deal about it, until a senior research suggests that
one would expect 12.5 significant tests even in the purely null case,
merely by chance. In that sense, finding only 11 significant results is
actually somewhat disappointing! … [Tukey] then proposed a sort of
<em>second-level significance testing</em>, … [to] indicate a kind of
<em>significance of the overall body of tests</em>.”</p>
</blockquote>
<p>All the tests performed in this package are designed to carry out the
sort of second-level significance testing suggested in the story. That
is, they are designed to test if there is at least one non-null
hypothesis in the entire group of hypotheses. In statistical terms,
assume that the researcher above had calculated 250 Z-scores, where
<span class="math inline">\(Z_{i}\)</span> has mean <span class="math inline">\(\mu_{i}\)</span> and variance 1. Then the GBJ null
hypothesis is that <span class="math inline">\(\mu_{i}=0\)</span> for
all <span class="math inline">\(i\)</span>, and the GBJ alternative is
that <span class="math inline">\(\mu_{i} \neq 0\)</span> for at least
one <span class="math inline">\(i\)</span>.</p>
<p>GBJ can be used to test either an entire collection of hypotheses, or
it may make more sense to partition the group into smaller, predefined
sets and then apply GBJ multiple times. For example, in the case of
GWAS, we can group the individual SNP test statistics into
genes/pathways and apply GBJ to each gene or pathway. In this case, GBJ
is testing if the entire gene has any association with the outcome.</p>
<p>Notable advantages to using GBJ are:</p>
<ol style="list-style-type: decimal">
<li><p>GBJ is a generalization of the Berk-Jones test, which is known to
be optimal - in a certain sense - for detecting rare and weak signals
when factors in a set are independent. This is clearly a very relevant
guarantee for the genetics setting. GBJ modifies Berk-Jones to provide
better finite sample rejection regions when factors in a set are
correlated.</p></li>
<li><p>Analytic calculation of p-values (no need for
permutation).</p></li>
<li><p>No tuning parameters. Standard inputs (similar to SKAT).
Reasonable to apply GBJ on sets ranging from 2 to 2000 factors.</p>
<ol style="list-style-type: lower-alpha">
<li>Theoretically there is no upper limit, but we are currently bound by
the numerical precision in standard C++ routines. There are plans to
remove this obstacle through arbitrary precision libraries.</li>
</ol></li>
</ol>
</div>
<div id="worked-example---individual-level-genotype-data" class="section level2">
<h2>Worked Example - Individual Level Genotype Data</h2>
<p>Suppose we are interested in testing whether a specific gene is
associated with pancreatic cancer. We have 1000 patients in our study,
half with pancreatic cancer and half without. Our dataset consists of 50
SNPs in the gene of interest, and for each patient we have their minor
allele count (0,1,2) at each of the 50 SNPs. Additionally we have
information on each patient’s age and gender:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(GBJ)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">0</span>)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a>cancer_status <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="dv">1</span>,<span class="dv">500</span>), <span class="fu">rep</span>(<span class="dv">0</span>,<span class="dv">500</span>))</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="co"># All of our SNPs have minor allele frequency of 0.3 in this example</span></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a>genotype_data <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="at">data=</span><span class="fu">rbinom</span>(<span class="at">n=</span><span class="dv">1000</span><span class="sc">*</span><span class="dv">50</span>, <span class="at">size=</span><span class="dv">2</span>, <span class="at">prob=</span><span class="fl">0.3</span>), <span class="at">nrow=</span><span class="dv">1000</span>)</span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a>age <span class="ot">&lt;-</span> <span class="fu">round</span>( <span class="fu">runif</span>(<span class="at">n=</span><span class="dv">1000</span>, <span class="at">min=</span><span class="dv">30</span>, <span class="at">max=</span><span class="dv">80</span>) )</span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a>gender <span class="ot">&lt;-</span> <span class="fu">rbinom</span>(<span class="at">n=</span><span class="dv">1000</span>, <span class="at">size=</span><span class="dv">1</span>, <span class="at">prob=</span><span class="fl">0.5</span>)     <span class="co"># Let 1 denote a female and 0 a male</span></span></code></pre></div>
<p>Under the null hypothesis of no association between gene and
pancreatic cancer, we can assume the true logistic model to be:</p>
<p><span class="math display">\[\text{logit}(\mu_{i}) = \beta_{0} +
\beta_{1}*Age_{i} + \beta_{2}*Gender_{i}\]</span></p>
<p>(actually since we generated the data, we know <span class="math inline">\(\beta_{1}=\beta_{2}=0\)</span>, but this is just
for illustration)</p>
<p>The function <code>calc_score_stats()</code> can be used to calculate
score statistics for each of the 50 SNPs. Under the null, these
statistics will have an asymptotic N(0,1) distribution.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>null_mod <span class="ot">&lt;-</span> <span class="fu">glm</span>(cancer_status<span class="sc">~</span>age<span class="sc">+</span>gender, <span class="at">family=</span><span class="fu">binomial</span>(<span class="at">link=</span><span class="st">&quot;logit&quot;</span>))</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>log_reg_stats <span class="ot">&lt;-</span> <span class="fu">calc_score_stats</span>(<span class="at">null_model=</span>null_mod, <span class="at">factor_matrix=</span>genotype_data, <span class="at">link_function=</span><span class="st">&quot;logit&quot;</span>)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>log_reg_stats<span class="sc">$</span>test_stats</span></code></pre></div>
<pre><code>##  [1] -0.68383391 -1.52282289 -0.04324001  1.02900792 -2.08636456 -0.47750816
##  [7]  1.02134595  0.57157512  0.09543809  1.88826217 -0.50314642 -0.39237040
## [13] -1.00903095  0.70567669 -0.74408222  1.17911250  0.63650614 -1.45052034
## [19] -0.04092308  1.06692697 -1.39140523 -0.15765491  0.11381805  0.44208014
## [25] -0.62752319  0.77393856 -0.06840144 -0.27651351  0.03332305  0.96798877
## [31]  0.65627672  0.18681278 -1.11695309 -1.58261616  1.44547339  1.78314038
## [37]  0.88944355  0.44694854 -1.34907210  1.02299625 -0.59392262 -0.87602299
## [43]  0.55980215 -1.39571015 -0.16295461 -0.86944840 -0.21030648 -1.61408239
## [49]  0.64425823  1.71642921</code></pre>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>log_reg_stats<span class="sc">$</span>cor_mat[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>##              [,1]        [,2]        [,3]        [,4]         [,5]
## [1,]           NA  0.02020415 -0.01580513  0.03906158  0.005420753
## [2,]  0.020204152          NA -0.03967446  0.03321533  0.028975255
## [3,] -0.015805130 -0.03967446          NA -0.02172816  0.034011177
## [4,]  0.039061583  0.03321533 -0.02172816          NA -0.043455163
## [5,]  0.005420753  0.02897526  0.03401118 -0.04345516           NA</code></pre>
<p>If our outcome was continuous and we wanted to assume a linear
regression model, then we could still use
<code>calc_score_stats()</code> with
<code>link_function=&#39;linear&#39;</code>, or if the outcome was non-negative
count data and we assume a Poisson regression model, then use
<code>link_function=&#39;log&#39;</code>.</p>
<p>Now we have both the test statistics and their correlation matrix, we
can apply GBJ.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>cor_Z <span class="ot">&lt;-</span> log_reg_stats<span class="sc">$</span>cor_mat</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>score_stats <span class="ot">=</span> log_reg_stats<span class="sc">$</span>test_stats</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a><span class="fu">GBJ</span>(<span class="at">test_stats=</span>score_stats, <span class="at">cor_mat=</span>cor_Z)</span></code></pre></div>
<pre><code>## $GBJ
## [1] 0.6109885
## 
## $GBJ_pvalue
## [1] 0.6091488
## 
## $err_code
## [1] 0</code></pre>
<p>And that’s it! Now you are not convinced that GBJ is the correct test
for your application, you can also apply GHC, HC, BJ, or minP, as
demonstrated below:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="fu">GHC</span>(<span class="at">test_stats=</span>score_stats, <span class="at">cor_mat=</span>cor_Z)</span></code></pre></div>
<pre><code>## $GHC
## [1] 1.505614
## 
## $GHC_pvalue
## [1] 0.6328255
## 
## $err_code
## [1] 0</code></pre>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="fu">HC</span>(<span class="at">test_stats=</span>score_stats, <span class="at">cor_mat=</span>cor_Z)</span></code></pre></div>
<pre><code>## $HC
## [1] 1.514436
## 
## $HC_pvalue
## [1] 0.633645</code></pre>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="fu">BJ</span>(<span class="at">test_stats=</span>score_stats, <span class="at">cor_mat=</span>cor_Z)</span></code></pre></div>
<pre><code>## $BJ
## [1] 0.6208709
## 
## $BJ_pvalue
## [1] 0.607419</code></pre>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="fu">minP</span>(<span class="at">test_stats=</span>score_stats, <span class="at">cor_mat=</span>cor_Z)</span></code></pre></div>
<pre><code>## $minP
## [1] 0.03694561
## 
## $minP_pvalue
## [1] 0.8439997</code></pre>
</div>
<div id="worked-example---summary-statistics" class="section level2">
<h2>Worked Example - Summary Statistics</h2>
<p>Suppose now that we only have GWAS summary statistics, not
individual-level data, but we still want to perform a set-level test
using GBJ. We then need to estimate the correlations between these
summary statistics using genotypes from a reference panel. We have
provided a function, estimate_ss_cor() to perform the estimation.
estimate_ss_cor() requires as input (1) a matrix of <span class="math inline">\(m\)</span> PCs calculated from a reference panel
(of the same ethnicity) and (2) a matrix of the genotypes at the summary
statistic SNPs from the same reference panel. Here <span class="math inline">\(m\)</span> is approximately the number of PCs used
in the original analysis that produced the summary statistics.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a><span class="co"># Load the genotype data at FGFR2 SNPs for 91 Great Britain (GBR) subjects from the 1000 Genomes Project (publically available).</span></span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a><span class="fu">data</span>(FGFR2)</span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a></span>
<span id="cb16-4"><a href="#cb16-4" tabindex="-1"></a><span class="co"># Load PCs for these same 91 subjects (calculated, for example, with EIGENSTRAT)</span></span>
<span id="cb16-5"><a href="#cb16-5" tabindex="-1"></a><span class="fu">data</span>(gbr_pcs)</span>
<span id="cb16-6"><a href="#cb16-6" tabindex="-1"></a></span>
<span id="cb16-7"><a href="#cb16-7" tabindex="-1"></a><span class="co"># Suppose we were given the following 64 test statistics for the FGFR2 SNPs (must be the same SNPs that</span></span>
<span id="cb16-8"><a href="#cb16-8" tabindex="-1"></a><span class="co"># are in our genotype matrix!)</span></span>
<span id="cb16-9"><a href="#cb16-9" tabindex="-1"></a>FGFR2_stats <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="at">n=</span><span class="dv">64</span>)</span>
<span id="cb16-10"><a href="#cb16-10" tabindex="-1"></a></span>
<span id="cb16-11"><a href="#cb16-11" tabindex="-1"></a><span class="co"># Estimate correlation matrix for summary statistics</span></span>
<span id="cb16-12"><a href="#cb16-12" tabindex="-1"></a>FGFR2_cor_mat <span class="ot">&lt;-</span> <span class="fu">estimate_ss_cor</span>(<span class="at">ref_pcs=</span>gbr_pcs, <span class="at">ref_genotypes=</span>FGFR2, <span class="at">link_function=</span><span class="st">&#39;logit&#39;</span>)</span>
<span id="cb16-13"><a href="#cb16-13" tabindex="-1"></a></span>
<span id="cb16-14"><a href="#cb16-14" tabindex="-1"></a><span class="co"># Run GBJ</span></span>
<span id="cb16-15"><a href="#cb16-15" tabindex="-1"></a><span class="fu">GBJ</span>(<span class="at">test_stats=</span>FGFR2_stats, <span class="at">cor_mat=</span>FGFR2_cor_mat)</span></code></pre></div>
<pre><code>## $GBJ
## [1] 0.3366795
## 
## $GBJ_pvalue
## [1] 0.6390038
## 
## $err_code
## [1] 0</code></pre>
</div>
<div id="some-advice-for-special-cases" class="section level2">
<h2>Some advice for special cases</h2>
<ul>
<li><p>Note that in the first example (individual-level data) we know
each of our factors (SNPs) are independent, so another option would be
to just input a correlation matrix where all the off-diagonal elements
are zero. This should not drastically change the results since our
estimated correlations are so close to 0 anyway.</p></li>
<li><p>Limits on the mathematical precision of R and C++ currently limit
us somewhat for very large sets and very small p-values. There are plans
to remove these limitations in future iterations of the software by
using arbitrary precision libraries, however this has not yet been
tested. At the moment, we limit sets to 2000 factors at most. Also
p-values less than <span class="math inline">\(1*10^{(-14)}\)</span> are
generally rounded to this value.</p></li>
</ul>
<p>Questions or novel applications? Please let me know! Contact
information can be found in the package description.</p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
